lib/segment/src/index/hnsw_index/gpu/gpu_graph_builder.rs

```diff
@@ -90,9 +78,27 @@ pub fn build_hnsw_on_gpu<'a>(
         }
     }
 
+    // Check if all points are linked on CPU.
+    // If there are no batches left, we can return result before gpu resources creation.
+    if batched_points.iter_batches(cpu_linked_points_count).count() == 0 {
```

Comment:
Let's do `.next().is_none()` instead of count, so we don't have to exhaust the whole iterator.

```suggestion
    if batched_points
        .iter_batches(cpu_linked_points_count)
        .next()
        .is_none()
    {
```

Reply:
Thanks!